src/qdlutils/applications/qdlscan/main.py

```diff
@@ -0,0 +1,1289 @@
+            n_pixels = self.scan_parameters['line_pixels'],
+            time = self.scan_parameters['line_time'],
+            id = str(self.number_scans).zfill(3)
+        )
+
+    def optimize_y_axis(self, tkinter_event=None) -> None:
+        '''
+        Callback to optimize Y button in launcher GUI.
+        Gets the config data written in the GUI. If it is valid then it will create an 
+        instnace of the `LineScanApplication` class to scan the axis.
+
+        Parameters
+        ----------
+        tkinter_event: tk.Event
+            The button press event, not used.
+        '''
+        if self.application_controller.busy:
+            logger.error(f'Application controller is current busy.')
+            return None
+        logger.info('Optimizing Y axis.')
+        # Update the parameters
+        try:
+            self._get_scan_config()
+        except Exception as e:
+            logger.error(f'Scan parameters are invalid: {e}')
+            return None
+        # Increase the nunmber of scans launched
+        self.number_scans += 1
+        # Launch a line scan application
+        self.current_scan = LineScanApplication(
+            parent_application = self,
+            application_controller = self.application_controller,
+            axis = 'y',
+            range = self.scan_parameters['line_range_xy'],
+            n_pixels = self.scan_parameters['line_pixels'],
+            time = self.scan_parameters['line_time'],
+            id = str(self.number_scans).zfill(3)
+        )
+
+    def optimize_z_axis(self, tkinter_event=None) -> None:
+        '''
+        Callback to optimize Z button in launcher GUI.
+        Gets the config data written in the GUI. If it is valid then it will create an 
+        instnace of the `LineScanApplication` class to scan the axis.
+
+        Parameters
+        ----------
+        tkinter_event: tk.Event
+            The button press event, not used.
+        '''
+        if self.application_controller.busy:
+            logger.error(f'Application controller is current busy.')
+            return None
+        logger.info('Optimizing Z axis.')
+        # Update the parameters
+        try:
+            self._get_scan_config()
+        except Exception as e:
+            logger.error(f'Scan parameters are invalid: {e}')
+            return None
+        # Increase the nunmber of scans launched
+        self.number_scans += 1
+        # Launch a line scan application
+        self.current_scan = LineScanApplication(
+            parent_application = self,
+            application_controller = self.application_controller,
+            axis = 'z',
+            range = self.scan_parameters['line_range_z'],
+            n_pixels = self.scan_parameters['line_pixels'],
+            time = self.scan_parameters['line_time'],
+            id = str(self.number_scans).zfill(3)
+        )
+
+    def start_image_scan(self, tkinter_event=None) -> None:
```

Comment:
at the end of the scan, put blue circle in the middle of the scan which should correspond to the actual position

Reply:
it would also be worth it to add a marker for the prior position that is not the blue circle when you move positions by right clicking "go to position"

Reply:
Updated so that blue circle position is updated to final position after scan is complete.

I will not implement a "prior position" marker in this PR, but this would make a good first issue for new developers.